packages/grafana-ui/src/types/icon.ts

```diff
@@ -1,9 +1,23 @@
+import React from 'react';
 import { ComponentSize } from './size';
-export type IconType = 'mono' | 'default';
+
+// Exports of main Icon types
+export type IconProps = SolidIconProps | DefaultIconProps;
+export type IconType = DefaultIconType | SolidIconType;
+export type IconName = DefaultIconName | SolidIconName;
 export type IconSize = ComponentSize | 'xl' | 'xxl' | 'xxxl';
 
-export type IconName =
+// Default types
+export interface DefaultIconProps extends React.HTMLAttributes<HTMLDivElement> {
+  name: DefaultIconName;
+  type?: DefaultIconType;
```

Comment:
since we have only two types of the icon I suggest using boolean flag instead of `type`:`solid: boolean`